ovh/resource_cloud_project_database_log_subscription.go

```diff
@@ -0,0 +1,222 @@
+				Description: "Log kind name of this subscription",
+				Computed:    true,
+			},
+			"ldp_service_name": {
+				Type:        schema.TypeString,
+				Description: "Name of the destination log service",
+				Sensitive:   true,
+				Computed:    true,
+			},
+			"operation_id": {
+				Type:        schema.TypeString,
+				Description: "Identifier of the operation",
+				Computed:    true,
+			},
+			"resource_name": {
+				Type:        schema.TypeString,
+				Description: "Name of subscribed resource, where the logs come from",
+				Computed:    true,
+			},
+			"resource_type": {
+				Type:        schema.TypeString,
+				Description: "Type of subscribed resource, where the logs come from",
+				Computed:    true,
+			},
+			"updated_at": {
+				Type:        schema.TypeString,
+				Description: "Last update date of the subscription",
+				Computed:    true,
+			},
+		},
+	}
+}
+
+func resourceCloudProjectDatabaseLogSubscriptionImportState(d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	givenID := d.Id()
+	n := 4
+	splitID := strings.SplitN(givenID, "/", n)
+	if len(splitID) != n {
+		return nil, fmt.Errorf("import Id is not service_name/engine/cluster_id/id formatted")
+	}
+	serviceName := splitID[0]
+	engine := splitID[1]
+	clusterID := splitID[2]
+	id := splitID[3]
+	d.SetId(id)
+	d.Set("cluster_id", clusterID)
+	d.Set("engine", engine)
+	d.Set("service_name", serviceName)
+
+	results := make([]*schema.ResourceData, 1)
+	results[0] = d
+	return results, nil
+}
+
+func resourceCloudProjectDatabaseLogSubscriptionCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*Config)
+	serviceName := d.Get("service_name").(string)
+	engine := d.Get("engine").(string)
+	clusterID := d.Get("cluster_id").(string)
+
+	endpoint := fmt.Sprintf("/cloud/project/%s/database/%s/%s/log/subscription",
+		url.PathEscape(serviceName),
+		url.PathEscape(engine),
+		url.PathEscape(clusterID),
+	)
+	params := (&CloudProjectDatabaseLogSubscriptionCreateOpts{}).fromResource(d)
+	res := &CloudProjectDatabaseLogSubscriptionResponse{}
+
+	log.Printf("[DEBUG] Will create Log subscrition : %+v for cluster %s from project %s", params, clusterID, serviceName)
+	err := config.OVHClient.Post(endpoint, params, res)
+	if err != nil {
+		diag.Errorf("calling Post %s with params %+v:\n\t %q", endpoint, params, err)
+	}
+
+	log.Printf("[DEBUG] Waiting for Log subscription operation %s to be READY", res.OperationID)
+	op, err := waitForDbaasLogsOperation(ctx, config.OVHClient, res.LDPServiceName, res.OperationID)
+	if err != nil {
+		return diag.Errorf("timeout while waiting log subscrition operation %s to be READY: %q", res.OperationID, err)
+	}
+	log.Printf("[DEBUG] Log subscrition operation %s is READY", res.OperationID)
+
+	d.SetId(*op.SubscriptionID)
+
+	readDiags := resourceCloudProjectDatabaseLogSubscriptionRead(ctx, d, meta)
+	err = diagnosticsToError(readDiags)
+	if err != nil {
```

Comment:
why don't you use `readDiags.HasErrors()` instead to avoid `diagnosticsToError` and then `diag.FromErr(err)` ?

Reply:
Yes even `return resourceCloudProjectDatabaseLogSubscriptionRead(ctx, d, meta)`